src/cunumeric/device_scalar_reduction_buffer.h

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "core/cuda/cuda_help.h"
+#include "core/data/buffer.h"
+
+namespace cunumeric {
+
+template <typename REDOP>
+class DeviceScalarReductionBuffer {
+ private:
+  using VAL = typename REDOP::RHS;
+
+ public:
+  DeviceScalarReductionBuffer(cudaStream_t stream)
+    : buffer_(legate::create_buffer<VAL>(1, Legion::Memory::Kind::GPU_FB_MEM))
+  {
+    VAL identity{REDOP::identity};
+    ptr_ = buffer_.ptr(0);
+    CHECK_CUDA(cudaMemcpyAsync(ptr_, &identity, sizeof(VAL), cudaMemcpyHostToDevice, stream));
+  }
+
+  template <bool ATOMIC>
```

Comment:
I would prefer to keep the meaning of this boolean flag consistent with Legion's, i.e. use "exclusive" here instead of "atomic". Especially since we use `Reduction::fold` operators in other places in the code (which takes "exclusive"), this is a misunderstanding waiting to happen.

Reply:
https://github.com/nv-legate/cunumeric/pull/509/commits/7d4324647b69f45bd493b16ac25185a3e4d36100